crates/sui-metrics-proxy/src/main.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use clap::Parser;
+use std::net::SocketAddr;
+use sui_metrics_proxy::admin::{app, set_tracing, shutdown_signal};
+use tracing::info;
+
+const GIT_REVISION: &str = {
+    if let Some(revision) = option_env!("GIT_REVISION") {
+        revision
+    } else {
+        let version = git_version::git_version!(
+            args = ["--always", "--dirty", "--exclude", "*"],
+            fallback = ""
+        );
+
+        if version.is_empty() {
+            panic!("unable to query git revision");
+        }
+        version
+    }
+};
+const VERSION: &str = const_str::concat!(env!("CARGO_PKG_VERSION"), "-", GIT_REVISION);
+
+#[derive(Parser, Debug)]
+#[clap(rename_all = "kebab-case")]
+#[clap(name = env!("CARGO_BIN_NAME"))]
+#[clap(version = VERSION)]
+struct Args {
+    #[clap(long, short, help = "Specify address to listen on")]
+    listen_address: SocketAddr,
+    #[clap(long, short, help = "Specify an upstream address to send to")]
+    upstream_address: SocketAddr,
+    #[clap(long, short, help = "mpsc buffer size - ideally set above our max rps")]
+    buffer_size: usize,
```

Comment:
I would make sure there is a default for this, so that we don't need to set it every time we invoke this.

Reply:
good point. i can do that. In my tests i picked 10,000.